### ISSUE TITLE
UI: Global Statusline

### ISSUE THREAD

**Issue by LunarLambda** [neovim/neovim#9342]
## What we have right now

- Every window has its own statusline
- But: Only one can be active at once
- Inactive statusline gives no useful information...
- Any sufficiently sophisticated statusline (powerline, etc) will get truncated in vertical splits
- Truncation increases visual clutter
- The default powerline setup easily fills up 80 characters (without plugins):
![image](https://user-images.githubusercontent.com/38919842/49722881-fa895200-fc65-11e8-9ebd-2317b7bd39df.png)

## The idea

Wouldn't it be nice to have a single statusline at the bottom, like in screen or tmux?

- Statusline is always in the same spot, 'where you expect it to be'
- We already have a 'global' command line: It's at the bottom!
- Less visual clutter, nicer window seperators (see below)
- Might speed up the UI: 'statusline' needs to be evaluated only for the active window, rather than all.
- Only restricted in size by your Terminal.

## How could it be done?

I'm strongly in favor of a 'laststatus' value of 3 enabling the global statusline, since any value < 0 or > 2 currently doesn't error, and instead behaves like 'laststatus' = 1.

Currently, only vertical window seperators are used. ('vert' value in 'fillchars')
All Terminals I've tried correctly display box drawing characters (tested with tmux), so I think those should be the default. Otherwise, window seperators would probably need their own option:

'windowseperators' (default: ─│├┤┬┴┼, for 1-1, 1|1, 1|2, 2|1 1-2, 2-1, and 2*2 layouts respectively).

Another thing that *could* be done would be to highlight the window seperators around the active window differently, like tmux does. This could be used to more easily distinguish the active window when the global statusline is enabled, although I don't think that is really necessary.

## Costs

Implementation: medium (Additional options and ui code)
Maintainability: minor

## A look into the future?

This mock-up was made with tmux as well as some slight editing:
![nvim-global-statusline-mock](https://user-images.githubusercontent.com/38919842/49723015-3de3c080-fc66-11e8-9a16-71c3cceedf56.png)

**Comment by bfredl** [comment]
To start with, we should allow this in  `ext_multigrid` mode. We should add a `ext_statusline` extension similar to `ext_tabline`. Later on, new flexible TUI can be implemented on top of multiline, that allows this and similar stuff (such as cmdline at the bottom of each window ).

**Comment by weilbith** [comment]
Was there ever any new update to this?

**Comment by caenrique** [comment]
I would love to se this feature implemented. Is this even planned??

**Comment by iasj** [comment]
This would on help my plugin development just like I posted [here](https://github.com/neovim/neovim/issues/1435#issuecomment-670837497).

**Comment by windwp** [comment]
my status line can do that it use a floating window on bottom
https://github.com/windwp/windline.nvim

**Comment by famiu** [comment]
> @windwp plugin is awesome, but using the floating window I saw some issues between plugins because most plugins presume that it won't be there. So GitDiff breaks, there is the issue with the last window being a floating window and other pain points. It seems like being built-in could keep all of those assumptions and avoid those weird integration errors.

I'm planning to work on a PR to implement a global statusline within Neovim soon, stay tuned for that.

**Comment by windwp** [comment]
@AZanellato It already has a PR to fix that issue  #15378 #14387. I already try and it working
I think it will merge on 0.6.

@famiu of course if you can make a PR to make neovim have a global status line then it will be better.

**Comment by vimpostor** [comment]
My statusline plugin can do this already btw: https://github.com/vimpostor/vim-tpipeline
It embeds the statusline in the tmux statusline, thus making it a global (external) statusline.

Right now it uses some obscure hackery to get the statusline, but once that the good work from @famiu in #16020 is merged, I will port it to the clean new `nvim_eval_statusline` API, which should make it work reliable with all statuslines.

**Comment by mjlbach** [comment]
Closed by #17266
